pkg/apis/toolchain/v1alpha1/nstemplateset_types.go

```diff
@@ -59,6 +59,10 @@ type NSTemplateSetNamespace struct {
 	// Optional field. Used to specify a custom template
 	// +optional
 	Template string `json:"template,omitempty"`
+
+	// TemplateRef the name of the TemplateTier resource on the host cluster and which contains the template to use
```

Comment:
I would use slightly different wording - it's (after all) used in the description withing the CRD. 
In addition, you were apparently influenced by my mistake with the name - it's not `TemplateTier` but `TierTemplate` :-) 
how about having something like this:
```suggestion
	// The name of the TierTemplate resource that exists in the host cluster and that contains the template to be used
```